py-*: obsolete py26/py33 subports in various Python ports

reneeotten opened the pull request:
#### Description
This PR contains another round of housekeeping on various Python ports in the tree by removing obsolete py26/py33 subports that have nothing depend on them anymore. No functional changes were made to these ports, so everything should still build.

In addition it contains the following updates to py26-only ports:
- py-logilab-hmm: add py27 subport
- py-pybox2d: update to 2.3.2, add py27 subport
- py26-pydb: obsolete port, replaced_by py-trepan (as stated by upstream)

<!-- Note: it is best make pull requests from a branch rather than from master -->

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
<!-- Generate version information with this command in shell:
    echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion)"; echo "Xcode $(xcodebuild -version | awk '{print $NF}' | tr '\n' ' ')"
-->
macOS 10.13.6 17G65
Xcode 9.4.1 9F2000

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?
<!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`? Only for ports where functional changes were made (i.e., py-logilab-hmm and py-pybox2d).
- [ ] tested basic functionality of all binary files?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

yan12125 commented on python/py-logilab-hmm/Portfile:
```diff
@@ -21,17 +23,19 @@ livecheck.url       http://ftp.logilab.org/pub/hmm/
-    universal_variant   no
-    python.add_archflags no
+    universal_variant       no
+    python.add_archflags    no
```
Inconsistent whitespacing here

reneeotten replied:
what is inconsistent here - the whitespacing is multiples of four and it's aligned within the ```if {${subport} ne ${name}}```. What would you like to be changed?

yan12125 replied:
Sorry I shouldn't read codes when sleepy :/

yan12125 commented on python/py-pybox2d/Portfile:
```diff
@@ -14,20 +15,19 @@ long_description    ${description}
-                    rmd160  362a16da2932f1bc748696f384649b6641195e53
-
-python.versions     26
+python.versions     27
```
Looks like pybox2d support Python 3.x. From https://github.com/pybox2d/pybox2d/blob/master/INSTALL.md:

> Recent builds should be available for Windows, Linux, and OS X, with Python 2.7, 3.3, 3.4 and 3.5.

reneeotten replied:
sure, py36/py37 added

michaelld commented on python/py-logilab-hmm/Portfile:
```diff
@@ -21,17 +23,19 @@ livecheck.url       http://ftp.logilab.org/pub/hmm/
-checksums           md5     81f63768df022714e66f6d471245800a \
-                    sha1    6be2012ab3b7ed6ceac42b48a821f6e194c71f6f \
-                    rmd160  b62f3ac49e4d2681138e55490430c4bdfdf418f6
+checksums           rmd160  b62f3ac49e4d2681138e55490430c4bdfdf418f6 \
```
Please provide separate commits for updates and obsolesce.

yan12125 replied:
Sorry, I shouldn't read codes when sleepy :/

michaelld commented on python/py-pybox2d/Portfile:
```diff
@@ -14,20 +15,19 @@ long_description    ${description}
 
-homepage            http://pybox2d.googlecode.com/
-master_sites        googlecode:pybox2d
+checksums           rmd160  f40390a1567754c6c26497b5d3d9f6ba9d0a626b \
```
Please provide separate commits for updates and obsolesce.

michaelld commented on science/Gyoto/Portfile:
```diff
@@ -137,7 +143,7 @@ if {[string match libc++ ${configure.cxx_stdlib}]} {
 For multi-processing, wrap gyoto in mpirun-openmpi-${mpi_suffix}, e.g.:
   $ mpirun-openmpi-${mpi_suffix} gyoto -P4 in.xml out.fits
-"    
+"
```
Please provide separate commits for whitespace and obsolesce.

michaelld commented:
Thank you for the PR; we need work like this! That said, please provide separate commits / PRs for things that are distinctly separable, such as port updates, whitespace, and obsolesce. Once the various parts of this PR are split & this one updated to just be obsolesce, I will certainly approve it!

reneeotten commented:
@michaelld thanks for the comments - I have updated the commits as requested. All commits in this PR were made with the intention to remove py26/py33 subports. I just updated a few of them to their latest version and/or made sure they would still build. I think that should be fine in one PR, the commits should now make that clear.

michaelld commented:
@reneeotten Yeah there's no one correct way to do PRs. Sometimes we have to mix & match a little to get the end result to work. I'm fine with what you've done here; just wanted to point out what's preferred when reasonably possible. Thx for addressing the commits in this manner. LGTM now.

pmetzger commented on python/py-chardet/Portfile:
```diff
@@ -18,7 +18,7 @@ long_description    Character encoding auto-detection in Python. \
                     sha256  ad813ca7b31b19e12c506869ceec53dbd2b9ed5fa73955895394e0245ffe70a2
 
-python.versions     26 27 33 34 35 36 37
+python.versions     26 27 34 35 36 37
```
Is leaving 26 here intentional?

reneeotten replied:
yes it is for now: the ```canto``` port depends on it. It seems though that this would benefit from an update (see: https://trac.macports.org/ticket/49776), but I haven't looked into this.

pmetzger commented:
Overall LGTM.

g5pw commented:
LGTM for `py-six` 👍

pmetzger commented:
Merged. Thanks, @reneeotten!
